samples/extensions/dynamic_multisample_rasterization/dynamic_multisample_rasterization.cpp

```diff
@@ -0,0 +1,921 @@
+
+	VkImageSubresourceRange depth_range{range};
+	depth_range.aspectMask = VK_IMAGE_ASPECT_DEPTH_BIT;
+
+	for (int32_t i = 0; i < draw_cmd_buffers.size(); ++i)
+	{
+		VK_CHECK(vkBeginCommandBuffer(draw_cmd_buffers[i], &command_buffer_begin_info));
+
+		if (sample_count != VK_SAMPLE_COUNT_1_BIT)
+		{
+			attachments[0].resolveImageView = swapchain_buffers[i].view;
+		}
+		else
+		{
+			attachments[0].imageView = swapchain_buffers[i].view;
+		}
+		auto render_area             = VkRect2D{VkOffset2D{}, VkExtent2D{width, height}};
+		auto render_info             = vkb::initializers::rendering_info(render_area, 1, &attachments[0]);
+		render_info.layerCount       = 1;
+		render_info.pDepthAttachment = &attachments[1];
+
+		vkb::image_layout_transition(draw_cmd_buffers[i],
+		                             swapchain_buffers[i].image,
+		                             VK_PIPELINE_STAGE_COLOR_ATTACHMENT_OUTPUT_BIT,
+		                             VK_PIPELINE_STAGE_COLOR_ATTACHMENT_OUTPUT_BIT,
+		                             0,
+		                             VK_ACCESS_COLOR_ATTACHMENT_WRITE_BIT,
+		                             VK_IMAGE_LAYOUT_UNDEFINED,
+		                             VK_IMAGE_LAYOUT_COLOR_ATTACHMENT_OPTIMAL,
+		                             range);
+
+		vkb::image_layout_transition(draw_cmd_buffers[i],
+		                             depth_stencil.image,
+		                             VK_IMAGE_LAYOUT_UNDEFINED,
+		                             VK_IMAGE_LAYOUT_DEPTH_ATTACHMENT_OPTIMAL,
+		                             depth_range);
+
+		vkCmdBeginRenderingKHR(draw_cmd_buffers[i], &render_info);
+		vkCmdSetRasterizationSamplesEXT(draw_cmd_buffers[i], sample_count);        // VK_EXT_extended_dynamic_state3
+
+		VkViewport viewport = vkb::initializers::viewport(static_cast<float>(width), static_cast<float>(height), 0.0f, 1.0f);
+		vkCmdSetViewport(draw_cmd_buffers[i], 0, 1, &viewport);
+
+		VkRect2D scissor = vkb::initializers::rect2D(width, height, 0, 0);
+		vkCmdSetScissor(draw_cmd_buffers[i], 0, 1, &scissor);
+
+		vkCmdBindDescriptorSets(draw_cmd_buffers[i], VK_PIPELINE_BIND_POINT_GRAPHICS, pipeline_layout, 0, 1, &descriptor_set, 0, nullptr);
+
+		vkCmdBindPipeline(draw_cmd_buffers[i], VK_PIPELINE_BIND_POINT_GRAPHICS, pipeline_opaque);
+
+		for (auto &node : scene_nodes_opaque)
+		{
+			draw_node(draw_cmd_buffers[i], node);
+		}
+
+		vkCmdBindPipeline(draw_cmd_buffers[i], VK_PIPELINE_BIND_POINT_GRAPHICS, pipeline_opaque_flipped);
+
+		for (auto &node : scene_nodes_opaque_flipped)
+		{
+			draw_node(draw_cmd_buffers[i], node);
+		}
+
+		vkCmdBindPipeline(draw_cmd_buffers[i], VK_PIPELINE_BIND_POINT_GRAPHICS, pipeline_transparent);
+
+		for (auto &node : scene_nodes_transparent)
+		{
+			draw_node(draw_cmd_buffers[i], node);
+		}
+
+		vkCmdBindPipeline(draw_cmd_buffers[i], VK_PIPELINE_BIND_POINT_GRAPHICS, pipeline_transparent_flipped);
+
+		for (auto &node : scene_nodes_transparent_flipped)
+		{
+			draw_node(draw_cmd_buffers[i], node);
+		}
```

Comment:
Before binding a pipeline, one could check if there are nodes for that pipeline at all, like
```
		if (!scene_nodes_opaque.empty())
		{
			vkCmdBindPipeline(draw_cmd_buffers[i], VK_PIPELINE_BIND_POINT_GRAPHICS, pipeline_opaque);
			for (auto &node : scene_nodes_opaque)
			{
				draw_node(draw_cmd_buffers[i], node);
			}
		}

```

Reply:
Changes applied.